spec/src/main/asciidoc/app-programming-model.adoc

```diff
@@ -155,9 +155,62 @@ MAY stop the deployment of an application if Metadata is missing.
 
 [[annotated-naming-convention]]
 ==== Annotated Naming Convention
-Annotated metrics are registered into the _application_ `MetricRegistry` with the name based on the annotation's `name` and `absolute` fields.
+Annotated metrics are registered into the _application_ `MetricRegistry` with the name based on
 
-.Example of annotated metric names
+- The annotation's `name` parameter (herein referred to as "local name"),  
+- the annotation's `absolute` parameter and
+- the fully qualified (canonical) name of the declaring class.
+
+If the `name` parameter is not specified, then the local name will be the short name of the declaring element (name of the field or method).
+For constructors, the local name in this case is the short name of the class to which this constructor belongs.
```

Comment:
```suggestion
For constructors, the local name in this case is the simple name of the class to which this constructor belongs.
```
As the method is named `getSimpleName` it might be more clear?